src/lazy/encoder/text/mod.rs

```diff
@@ -50,7 +48,8 @@ impl<W: Write> LazyRawTextWriter_1_0<W> {
         TextValueWriter_1_0::new(
             self,
             0,
-            self.whitespace_config.space_between_top_level_values,
+            "", // No delimiter between values at the top level
```

Comment:
I'm missing something here. I don't see how the other changes allow us to have no delimiter between top level values here. I also don't know why the parent type here is `TopLevel`. I thought this _is_ the top level writer?

Reply:
> I'm missing something here. I don't see how the other changes allow us to have no delimiter between top level values here.

This code was adding a single space between top level values, but the parameter was actually expecting the caller to provide a _delimiter_ between values -- `,` for list/struct, or the empty string for sexp/top-level. Doing both resulted in too much whitespace in the output; not invalid, but not we wanted.

> I also don't know why the parent type here is TopLevel.

Each parent type (top level, list, sexp, struct) can construct a `*ValueWriter` that will encode the provided value into that context. Because this value writer is being created by the `LazyRawTextWriter_1_0` (and not by a container writer), it is going to write the value it encodes into the top level. Thus, the value's parent is the top level.